lua/entities/lucian_door_opener.lua

```diff
@@ -148,93 +170,79 @@ function ENT:Touch(ent)
 		ent.Door.Attached = true
-		self.Entity.IsMain = true
-		self.Entity.TargetDoor = ent.Door
-		self.Entity:SetPos(self.Entity.TargetDoor:LocalToWorld(Vector(0,0,0)))
-		self.Entity:SetAngles(self.Entity.TargetDoor:LocalToWorldAngles(Angle(90,0,0)))
-		self.Entity:SetCollisionGroup(COLLISION_GROUP_WORLD)
-		self.Entity:FollowBone(self.Entity.TargetDoor, self.Entity.TargetDoor:LookupBone("RightLock"))
-	--	constraint.Weld(self.Entity,self.Entity.TargetDoor,0,0,0,true)
-		constraint.NoCollide(self.Entity,self.Entity.TargetDoor,0,0)
+		self.IsMain = true
+		self.TargetDoor = ent.Door
+		self:SetPos(self.TargetDoor:LocalToWorld(Vector(0,0,0)))
+		self:SetAngles(self.TargetDoor:LocalToWorldAngles(Angle(90,0,0)))
+		self:SetCollisionGroup(COLLISION_GROUP_WORLD)
+		self:FollowBone(self.TargetDoor, self.TargetDoor:LookupBone("RightLock"))
+		constraint.NoCollide(self,self.TargetDoor,0,0)
 	end
 	
 end
 
-
 function ENT:Think()
 
-	if not IsValid(self.Entity.TargetDoor) then
-		self:Remove()
-		return
-	end
-	
-	if (self.GetTimer!=0 and CurTime()-self.GetTimer>0) then
-		local activator = self.Activator
-		if (IsValid(activator)) then
-			activator:Give("lucian_door_opener_wep")
-		end
+	if not IsValid(self.TargetDoor) then
 		self:RemoveBothLDO()
 		return
 	end
 	
-	if self.Entity.TargetDoor.Attached and not self.Entity.SecondAttached and not self.Entity.TargetDoor.LucianProcessDone then
+	if self.TargetDoor.Attached and not self.SecondAttached and not self.TargetDoor.LucianProcessDone then
 		local second = ents.Create("lucian_door_opener");
 		second:Spawn();
 		second:Activate();
-		self.Entity.SecondAttached = true
-		second.MainLDO = self.Entity
+		self.SecondAttached = true
+		second.MainLDO = self
 		second.IsMain = false
-		second.TargetDoor = self.Entity.TargetDoor
-		second:SetPos(self.Entity.TargetDoor:LocalToWorld(Vector(0,0,0)))
-		second:SetAngles(self.Entity.TargetDoor:LocalToWorldAngles(Angle(90,0,0)))
+		second.TargetDoor = self.TargetDoor
+		second:SetPos(self.TargetDoor:LocalToWorld(Vector(0,0,0)))
+		second:SetAngles(self.TargetDoor:LocalToWorldAngles(Angle(90,0,0)))
 		second:SetCollisionGroup(COLLISION_GROUP_WORLD)
-		second:FollowBone(self.Entity.TargetDoor, self.Entity.TargetDoor:LookupBone("LeftLock"))
-	--	constraint.Weld(second,self.Entity.TargetDoor,0,0,0,true)
-		constraint.NoCollide(second,self.Entity.TargetDoor,0,0)
-		self.Entity.SecondLucianDevice = second
-		self.Entity.SecondLucianDevice.Untouchable = true
-		self.Entity.TargetDoor.LucianProcessDone = true
+		second:FollowBone(self.TargetDoor, self.TargetDoor:LookupBone("LeftLock"))
+		second.Owner = second.MainLDO.Owner
+		constraint.NoCollide(second,self.TargetDoor,0,0)
+		self.SecondLucianDevice = second
+		self.SecondLucianDevice.Untouchable = true
+		self.TargetDoor.LucianProcessDone = true
 	end
-
-	--if self.Entity.TargetDoor.Attached and self.Entity.TargetDoor.CanDoAnim then
-		self.Entity:NextThink(CurTime());
-	--end
 	
+	self:NextThink(CurTime()+0.1);
```

Comment:
It follow bone, and in mp without this it was look very weird. Or maybe i'm bit confuse..